### PR TITLE
Complete get endpoint, needs testing <3

### PR DESCRIPTION
**Pull request by dajwilson**

### CODE REVIEW

File: src/main/java/com/tasks/taskapi/TaskApiApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.tasks.taskapi;
+
+import org.springframework.boot.SpringApplication;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/TaskApiApplication.java:3]: unused

File: src/main/java/com/tasks/taskapi/controller/TaskController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.tasks.taskapi.model.Task;
+import com.tasks.taskapi.repository.TaskRepository;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/controller/TaskController.java]: unused

File: src/main/java/com/tasks/taskapi/controller/TaskController.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+@Controller
+@RequestMapping("/tasks")
+public class TaskController {
+
+    @Autowired
+    private TaskService taskService;
+
+    @GetMapping("/get-all")
+    public ResponseEntity<List<Task>> getAllTasks() {
+       List<Task> books =  taskService.getAllTasks();
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/controller/TaskController.java]: variable name should be tasks

File: src/main/java/com/tasks/taskapi/model/Task.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    @NonNull
+    private String name;
+
+    @JsonProperty("priority")
+    @NonNull
+    private Integer priority;
+
+    @JsonProperty("inprogress")
+    @NonNull
+    private Boolean inprogress;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/model/Task.java]: probably don't need in progress and incomplete ? An enum could be good here xx

File: src/main/java/com/tasks/taskapi/repository/TaskRepository.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import com.tasks.taskapi.model.Task;
+import com.tasks.taskapi.utils.TaskRowMapper;
+
+import org.springframework.boot.CommandLineRunner;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/repository/TaskRepository.java]: is this used ?

File: src/main/java/com/tasks/taskapi/repository/TaskRepository.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.boot.SpringApplication;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/repository/TaskRepository.java]: same here

File: src/main/java/com/tasks/taskapi/repository/TaskRepository.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/repository/TaskRepository.java]: same here

File: src/main/java/com/tasks/taskapi/service/TaskService.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package com.tasks.taskapi.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
```
Comment (bethanlawx) [inline src/main/java/com/tasks/taskapi/service/TaskService.java]: unused

File: src/main/resources/data.sql [hunk 9]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO Tasks (id, name, priority, inprogress, incomplete) VALUES (1, 'Brush Teeth', 3, true, false);
+INSERT INTO Tasks (id, name, priority, inprogress, incomplete) VALUES (1, 'Sleep', 3, false, false);
```
Comment (bethanlawx) [inline src/main/resources/data.sql]: this can be simplified I think you only need INSERT INTO once

File: src/main/resources/schema.sql [hunk 10]
```diff
@@ -0,0 +1,7 @@
+Create Table Tasks (
```
Comment (bethanlawx) [inline src/main/resources/schema.sql]: capitalise create table ?

File: src/test/java/com/tasks/taskapi/TaskApiApplicationTests.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package com.tasks.taskapi;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class TaskApiApplicationTests {
```
Comment (bethanlawx) [inline src/test/java/com/tasks/taskapi/TaskApiApplicationTests.java]: was this autogenerated ?

**Review by bethanlawx — CHANGES_REQUESTED** [review]
proud of you <3

File: src/main/java/com/tasks/taskapi/TaskApiApplication.java (hunk 1 above)
Comment (dajwilson) [reply src/main/java/com/tasks/taskapi/TaskApiApplication.java:3]: you are wrong

**Review by bethanlawx — APPROVED** [review]

**Pull request merged**